### PR TITLE
3.0.0 documentation

### PR DESCRIPTION
**Pull request by mw-hrastega**
Documented the support for:
- Global tool configuration in freestyle and multi-configuration projects.
- Test filtering in freestyle, multi-configuration, and Pipeline projects.

Updated/added screenshots on the guide and example pages.

Rearranged the table to align with the Run MATLAB Tests UI.

Made improvements to the text wherever possible.

### CODE REVIEW

**Review by nbhoski — APPROVED** [review]
Changes looks good to Me

File: CONFIGDOC.md [hunk 1]
```diff
@@ -20,14 +22,13 @@ When you define an automated pipeline of tasks in Jenkins&trade;, whether in the
 You can use the web UI provided by Jenkins to configure the plugin in freestyle and multi-configuration projects. To run MATLAB or Simulink in a Pipeline project, see [Set Up Pipeline Project](#set-up-pipeline-project).
 
 ### Use MATLAB in Build
-Once you install the plugin, **Use MATLAB version** appears in the **Build Environment** section of the project configuration window.
+Once you install the plugin, **Use MATLAB version** appears in the **Build Environment** section of the project configuration window. Select **Use MATLAB version** to specify the MATLAB version you want to use in the build. You can select one of the MATLAB versions that have been registered as a Jenkins tool, or you can select **Custom** if you want to specify a different version. For more information about registering MATLAB as a tool, see [Register MATLAB as Jenkins Tool](#register-matlab-as-jenkins-tool).
 
+In this example, the list includes two registered tools as well as the option for specifying a custom installation. If you select **Custom**, then you must enter the full path to your desired MATLAB root folder in the **MATLAB root** box.
```
Comment (abhargav-mw) [inline CONFIGDOC.md:27]: Should  we add a screenshot for Custom location here as well?

Comment (mw-hrastega) [reply CONFIGDOC.md]: I suggest we do not add a screenshot for these reasons:

1) The text clearly describes what will happen in the UI.
2) We already have two screenshots on the page that display the MATLAB root box.
3) While screenshots are nice, they are difficult to maintain.

I just updated the text as follows to better describe the MATLAB root box. However, if you feel strongly about having a screenshot, please let me know.

"If you select Custom, a MATLAB root box appears in the UI. You must enter the full path to your desired MATLAB root folder in this box."

File: CONFIGDOC.md [hunk 2]
```diff
@@ -40,95 +41,120 @@ If the build agent already has your desired MATLAB on the path, then you are not
...
+This build step uses a default setting to run tests authored using the MATLAB Unit Testing Framework or Simulink Test&trade;. By default, if your code is organized into files and folders within a MATLAB project, then the plugin includes any test files in the project that have been labeled as `Test`. If your code does not leverage a MATLAB project or uses a MATLAB release before R2019a, then the plugin includes all tests in the project workspace, including its subfolders.
+ 
+With the **Run MATLAB Tests** build step, you can customize your test run using the exisiting options in the step configuration panel. For example, you can add folders to the MATLAB search path, have control over which tests to run, and generate different types of artifacts. If you do not select any of the check boxes, the tests still run, and test failures fail the build.
```
[the remark this answers is not in the document]
Comment (mw-hrastega) [reply CONFIGDOC.md]: Thanks, Nikhil. I updated the sentence to this. Is it fine?

"If you do not select any of the check boxes, all of the tests in your project run and test failures fail the build."

File: CONFIGDOC.md (hunk 1 above)
Comment (abhargav-mw) [reply CONFIGDOC.md]: Sounds good.

**Review by abhargav-mw — APPROVED** [review]
Looks good.

**Pull request merged**